### PR TITLE
z_en_ko OK

### PR DESCRIPTION
**Pull request by mzxrules**
also contains object_fa, object_km1, object_kw1 xmls. Due to ZAPD limitations, I can't correctly link km1/kw1 due to texture alignment issues

### CODE REVIEW

File: tools/ido_recomp/linux/7.1/text.c [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#include <stdio.h>
```
Comment (ProjectRevoTPP) [inline tools/ido_recomp/linux/7.1/text.c:1]: did you mean to commit this?

**Comment by Dragorn421** [comment]
So, is this safe to review (and merge) now? Commit a023245b6b39fb76fa0517f4dc4b246bae89d4da (contains extracted scenes and rooms) still exists

File: assets/xml/objects/object_km1.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      
+        <!-- 32-bit aligned textures. -->
+        <Texture Name="gKm1TLUT_0000FC" OutName="tlut_0000FC" Format="rgb5a1" Width="256" Height="1" Offset="0xFC" />
+        <Texture Name="gKm1TunicTex" OutName="tunic" Format="i8" Width="8" Height="8" Offset="0x2FC" />
+        <Texture Name="gKm1BootWrinkleTex" OutName="boot_wrinkle" Format="i8" Width="8" Height="8" Offset="0x33C" />
+        <Texture Name="gKm1Skin1Tex" OutName="skin1" Format="ci8" TlutOffset="0xFC" Width="8" Height="8" Offset="0x37C" />
+        <Texture Name="gKm1TunicFringeTex" OutName="tunic_fringe" Format="rgb5a1" Width="16" Height ="16" Offset="0x3BC" />
+        <Texture Name="gKm1UnusedHairTex" OutName="unused_hair" Format="ci8" TlutOffset="0xFC" Width="16" Height="32" Offset="0x5BC" />
+        <Texture Name="gKm1Ear1Tex" OutName="ear1" Format="ci8" TlutOffset="0xFC" Width="8" Height="8" Offset="0x7BC"/>
+        <Texture Name="gKm1HandTex" OutName="hand" Format="ci8" Width="8" Height="8" Offset="0x7FC" />
```
Comment (Dragorn421) [inline assets/xml/objects/object_km1.xml:31]:
```suggestion
        <Texture Name="gKm1HandTex" OutName="hand" Format="ci8" TlutOffset="0xFC" Width="8" Height="8" Offset="0x7FC" />
```

File: assets/xml/objects/object_kw1.xml [hunk 3]
```diff
@@ -1,48 +1,49 @@
...
+        <!-- Arm/neck texture-->
+        <Texture Name="gKw1Skin1Tex" OutName="skin1" Format="ci8" TlutOffset="0xFC" Width="8" Height="8" Offset="0x3FC" />
+        <Texture Name="gKw1TunicTex" OutName="tunic" Format="i8" Width="8" Height="8" Offset="0x43C" />
+        
+        <!-- Possibly a skin gradient texture -->
+        <Texture Name="object_kw1_Tex_00047C" OutName="tex_00047C" Format="ci8" TlutOffset="0xFC" Width="8" Height="8" Offset="0x47C" />
+        <Texture Name="gKw1UnusedEyeOpenTex" OutName="unused_eye_open" Format="rgb5a1" Width="32" Height="32" Offset="0x4BC"/>
+        
+        <Texture Name="gKw1HandTex" OutName="hand" Format="ci8" Width="8" Height="8" Offset="0xCBC" />
+        <Texture Name="gKw1BeltBuckleTex" OutName="belt_buckle" Format="ci8" Width="8" Height="8" Offset="0xCFC" />
```
Comment (Dragorn421) [inline assets/xml/objects/object_kw1.xml:35]:
```suggestion
        <Texture Name="gKw1HandTex" OutName="hand" Format="ci8" TlutOffset="0xFC" Width="8" Height="8" Offset="0xCBC" />
        <Texture Name="gKw1BeltBuckleTex" OutName="belt_buckle" Format="ci8" TlutOffset="0xFC" Width="8" Height="8" Offset="0xCFC" />
```

File: assets/xml/objects/object_kw1.xml [hunk 4]
```diff
@@ -1,48 +1,49 @@
...
-        <Texture Name="object_kw1_Tex_00160C" OutName="tex_0000160C" Format="ci8" Width="32" Height="32" Offset="0x160C" />
-        -->
+        <Texture Name="gKw1EarTex" OutName="ear" Format="ci8" Width="8" Height="8" Offset="0xF0C"/>
+        <Texture Name="gKw1EyeOpenTex" OutName="eye_open" Format="ci8" TlutOffset="0xD3C" Width="32" Height="32" Offset="0xF4C"/>
+        <Texture Name="gKw1EyeHalfTex" OutName="eye_half" Format="ci8" TlutOffset="0xD3C" Width="32" Height="32" Offset="0x1A0C"/>
+        <Texture Name="gKw1EyeClosedTex" OutName="eye_closed" Format="ci8" TlutOffset="0xD3C" Width="32" Height="32" Offset="0x1E0C"/>
+        <Texture Name="gKw1HairTex" OutName="hair" Format="ci8" Width="16" Height="32" Offset="0x134C" />
+        <Texture Name="gKw1Skin2Tex" OutName="skin2" Format="ci8" TlutOffset="0xD3C" Width="8" Height="8" Offset="0x154C" />
+        <Texture Name="gKw1HairlineTex" OutName="hairline" Format="ci8" Width="16" Height="8" Offset="0x158C" />
+        <Texture Name="gKw1MouthTex" OutName="mouth" Format="ci8" Width="32" Height="32" Offset="0x160C" />
```
Comment (Dragorn421) [inline assets/xml/objects/object_kw1.xml:45]:
```suggestion
        <Texture Name="gKw1EarTex" OutName="ear" Format="ci8" Width="8" Height="8" Offset="0xF0C" TlutOffset="0xD3C"/>
        <Texture Name="gKw1EyeOpenTex" OutName="eye_open" Format="ci8" TlutOffset="0xD3C" Width="32" Height="32" Offset="0xF4C"/>
        <Texture Name="gKw1EyeHalfTex" OutName="eye_half" Format="ci8" TlutOffset="0xD3C" Width="32" Height="32" Offset="0x1A0C"/>
        <Texture Name="gKw1EyeClosedTex" OutName="eye_closed" Format="ci8" TlutOffset="0xD3C" Width="32" Height="32" Offset="0x1E0C"/>
        <Texture Name="gKw1HairTex" OutName="hair" Format="ci8" Width="16" Height="32" Offset="0x134C"  TlutOffset="0xD3C"/>
        <Texture Name="gKw1Skin2Tex" OutName="skin2" Format="ci8" TlutOffset="0xD3C" Width="8" Height="8" Offset="0x154C" />
        <Texture Name="gKw1HairlineTex" OutName="hairline" Format="ci8" Width="16" Height="8" Offset="0x158C"  TlutOffset="0xD3C"/>
        <Texture Name="gKw1MouthTex" OutName="mouth" Format="ci8" Width="32" Height="32" Offset="0x160C"  TlutOffset="0xD3C"/>
```

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 5]
```diff
@@ -47,93 +70,1248 @@ static ColliderCylinderInit D_80A9A100 = {
...
+EnKoHead sHead[] = { { OBJECT_KM1, 0x06001890, NULL },
+                     { OBJECT_KW1, 0x06002C10, sKw1_eyes },
+                     { OBJECT_FA, gFaDL, sFa_eyes } };
+
+typedef struct {
+    s16 objectId;
+    FlexSkeletonHeader* flexSkeletonHeader;
+} EnKoSkeleton;
+
+static EnKoSkeleton sSkeleton[2] = { { OBJECT_KM1, 0x060000F0 }, { OBJECT_KW1, 0x060000F0 } };
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:94]:
```suggestion
static EnKoSkeleton sSkeleton[2] = { { OBJECT_KM1, gKm1Skel }, { OBJECT_KW1, gKw1Skel } };
```

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 6]
```diff
@@ -47,93 +70,1248 @@ static ColliderCylinderInit D_80A9A100 = {
...
+    EnKo* this = THIS;
+    switch (ENKO_TYPE) {
+        case ENKO_TYPE_CHILD_FADO:
+            if (gSaveContext.eventChkInf[4] & 1) {
+                return 0x10DA;
+            }
+            if (CHECK_QUEST_ITEM(QUEST_KOKIRI_EMERALD)) {
+                return 0x10D9;
+            }
+            return ((gSaveContext.infTable[11] & 0x80) != 0) ? 0x10D8 : 0x10D7;
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:257]:
```suggestion
            return (gSaveContext.infTable[11] & 0x80) ? 0x10D8 : 0x10D7;
```

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 7]
```diff
@@ -47,93 +70,1248 @@ static ColliderCylinderInit D_80A9A100 = {
...
+                              /* ENKO_TYPE_CHILD_3    */ { -10.0f, 10.0f, 10.0f, -10.0f, -30.0f },
+                              /* ENKO_TYPE_CHILD_4    */ { 0.0f, 0.0f, 0.0f, -10.0f, -20.0f },
+                              /* ENKO_TYPE_CHILD_5    */ { 0.0f, 0.0f, 0.0f, -20.0f, -20.0f },
+                              /* ENKO_TYPE_CHILD_6    */ { 0.0f, 0.0f, 0.0f, -10.0f, -20.0f },
+                              /* ENKO_TYPE_CHILD_7    */ { 10.0f, 10.0f, 10.0f, -60.0f, -20.0f },
+                              /* ENKO_TYPE_CHILD_8    */ { -10.0f, -10.0f, -20.0f, -30.0f, -30.0f },
+                              /* ENKO_TYPE_CHILD_9    */ { -10.0f, -10.0f, -10.0f, -40.0f, -40.0f },
+                              /* ENKO_TYPE_CHILD_10   */ { 0.0f, 0.0f, 0.0f, -10.0f, -20.0f },
+                              /* ENKO_TYPE_CHILD_11   */ { -10.0f, -10.0f, -20.0f, -30.0f, -30.0f },
+                              /* ENKO_TYPE_CHILD_FADO */ { 0.0f, 0.0f, 0.0f, -20.0f, -20.0f } };
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:579]: does a trailing comma help the formatting not indent everything to the right?
same with `D_80A9A730`

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 8]
```diff
@@ -47,93 +70,1248 @@ static ColliderCylinderInit D_80A9A100 = {
...
+        result = false;
+    }
+    return result;
+}
+
+s32 func_80A97D68(EnKo* this, GlobalContext* globalCtx) {
+    s16 arg3;
+
+    if (this->unk_1E8.unk_00 != 0) {
+        if ((this->skelAnime.animation == &D_06006A60) == false) {
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:625]:
```suggestion
        if (this->skelAnime.animation != &D_06006A60) {
```
that doesn't match? same with other `== false` checks

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 9]
```diff
@@ -47,93 +70,1248 @@ static ColliderCylinderInit D_80A9A100 = {
...
+s32 func_80A98124(EnKo* this, GlobalContext* globalCtx) {
+    func_80034F54(globalCtx, this->unk_2E4, this->unk_304, 16);
+    func_80034A14(&this->actor, &this->unk_1E8, 2, 4);
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ko/func_80A96F48.s")
+s32 func_80A98174(EnKo* this, GlobalContext* globalCtx) {
+    if (this->unk_1E8.unk_00 != 0) {
+        if (Animation_OnFrame(&this->skelAnime, 18.0f) != 0) {
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:724]:
```suggestion
        if (Animation_OnFrame(&this->skelAnime, 18.0f)) {
```

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 10]
```diff
@@ -47,93 +70,1248 @@ static ColliderCylinderInit D_80A9A100 = {
...
+    this->lookDist = info->lookDist;
+    this->lookDist += this->collider.dim.radius;
+    this->appearDist = info->appearDist;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ko/func_80A98034.s")
+// Used to fetch actor animation?
+s32 EnKo_GetForestQuestState2(EnKo* this) {
+    if (LINK_IS_ADULT) {
+        return (CHECK_QUEST_ITEM(QUEST_MEDALLION_FOREST)) ? ENKO_FQS_ADULT_SAVED : ENKO_FQS_ADULT_ENEMY;
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:1018]:
```suggestion
        return CHECK_QUEST_ITEM(QUEST_MEDALLION_FOREST) ? ENKO_FQS_ADULT_SAVED : ENKO_FQS_ADULT_ENEMY;
```

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
Thankfully we squash merge because this branch's history is severely messed up 😅 
Good job on this, only a few actors left, we're getting there!

File: spec [hunk 11]
```diff
@@ -5014,12 +5013,15 @@ beginseg
     name "object_km1"
     romalign 0x1000
     include "build/baserom/object_km1.o"
+    //include "build/assets/objects/object_km1/object_km1.o"
```
[the remark this answers is not in the document]
Comment (mzxrules) [reply spec:5016]: ZAPD needs to be updated to not always extract u64 sized textures.

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 12]
```diff
@@ -47,93 +70,1248 @@ static ColliderCylinderInit D_80A9A100 = {
...
     { 20, 46, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ko/func_80A96DB0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ko/func_80A96EC4.s")
+CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
+
+static void* sFa_eyes[] = { gFaEyeOpenTex, gFaEyeHalfTex, gFaEyeClosedTex, NULL };
+static void* sKw1_eyes[] = { 0x06000F4C, 0x06001A0C, 0x06001E0C, NULL };
```
[the remark this answers is not in the document]
Comment (mzxrules) [reply src/overlays/actors/ovl_En_Ko/z_en_ko.c:77]: Km1 and Kw1 are non-matching due to ZAPD issue

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 13]
```diff
@@ -47,93 +70,1248 @@ static ColliderCylinderInit D_80A9A100 = {
...
+
+typedef struct {
+    s16 objectId;
+    Gfx* dList;
+    void** eyeTextures;
+} EnKoHead;
+
+EnKoHead sHead[] = { { OBJECT_KM1, 0x06001890, NULL },
+                     { OBJECT_KW1, 0x06002C10, sKw1_eyes },
```
[the remark this answers is not in the document]
Comment (mzxrules) [reply src/overlays/actors/ovl_En_Ko/z_en_ko.c:86]: Km1 and Kw1 are non-matching due to ZAPD issue

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c (hunk 5 above)
Comment (mzxrules) [reply src/overlays/actors/ovl_En_Ko/z_en_ko.c:94]: Km1 and Kw1 are non-matching due to ZAPD issue

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c (hunk 12 above)
[the remark this answers is not in the document]
Comment (Dragorn421) [reply src/overlays/actors/ovl_En_Ko/z_en_ko.c:77]: May I sugest something like
```suggestion
static void* sKw1_eyes[] = { 0x06000F4C /* gKw1EyeOpenTex */, 0x06001A0C /* gKw1EyeHalfTex */, 0x06001E0C /* gKw1EyeClosedTex */, NULL };
```
then? hopefully the formatter doesn't freak out

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c (hunk 8 above)
Comment (mzxrules) [reply src/overlays/actors/ovl_En_Ko/z_en_ko.c:625]: non-matching. doing this eliminates an `xor sltiu` pair

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 14]
```diff
@@ -47,93 +70,1254 @@ static ColliderCylinderInit D_80A9A100 = {
...
+        sp40 = this->unk_1E8.unk_08;
+        Matrix_RotateX(BINANG_TO_RAD(sp40.y), MTXMODE_APPLY);
+        Matrix_RotateZ(BINANG_TO_RAD(sp40.x), MTXMODE_APPLY);
+        Matrix_Translate(-1200.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+    if (limbIndex == 8 || limbIndex == 9 || limbIndex == 12) {
+        rot->y += Math_SinS(this->unk_2E4[limbIndex]) * 200.0f;
+        rot->z += Math_CosS(this->unk_304[limbIndex]) * 200.0f;
+    }
+    return 0;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:1271]: return false

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 15]
```diff
@@ -47,93 +70,1254 @@ static ColliderCylinderInit D_80A9A100 = {
...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ko/func_80A99560.s")
+// Post limb draw
+void EnKo_PostLimbDraw(GlobalContext* globalCtx2, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx, Gfx** gfx) {
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:1275]: comment not neeed

**Review by fig02 — COMMENTED** [review]
jenkins is also not happy with this, think theres an asm file that needs to go

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 16]
```diff
@@ -47,93 +70,1254 @@ static ColliderCylinderInit D_80A9A100 = {
...
     { 20, 46, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ko/func_80A96DB0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ko/func_80A96EC4.s")
+CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
+
+static void* sFa_eyes[] = { gFaEyeOpenTex, gFaEyeHalfTex, gFaEyeClosedTex, NULL };
+static void* sKw1_eyes[] = { /* gKw1EyeOpenTex */ 0x06000F4C, /* gKw1EyeHalfTex */ 0x06001A0C,
```
[the remark this answers is not in the document]
Comment (fig02) [reply src/overlays/actors/ovl_En_Ko/z_en_ko.c:77]: nvm i meant s, and theyre already there. but yea use camel case with no underscore

**Review by fig02 — APPROVED** [review]

**Review by Dragorn421 — APPROVED** [review]

File: assets/xml/objects/object_km1.xml [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+        <Limb Name="gKm1SkelLimb_0000A8" LimbType="Standard" Offset="0xA8" />
+      
+        <Skeleton Name="gKm1Skel" Type="Flex" LimbType="Standard" Offset="0xF0" />
+      
+        <!-- 32-bit aligned textures. -->
+        <Texture Name="gKm1TLUT_0000FC" OutName="tlut_0000FC" Format="rgb5a1" Width="256" Height="1" Offset="0xFC" />
+        <Texture Name="gKm1TunicTex" OutName="tunic" Format="i8" Width="8" Height="8" Offset="0x2FC" />
+        <Texture Name="gKm1BootWrinkleTex" OutName="boot_wrinkle" Format="i8" Width="8" Height="8" Offset="0x33C" />
+        <Texture Name="gKm1Skin1Tex" OutName="skin1" Format="ci8" TlutOffset="0xFC" Width="8" Height="8" Offset="0x37C" />
+        <Texture Name="gKm1TunicFringeTex" OutName="tunic_fringe" Format="rgb5a1" Width="16" Height ="16" Offset="0x3BC" />
```
Comment (Roman971) [inline assets/xml/objects/object_km1.xml:28]:
```suggestion
        <Texture Name="gKm1TunicFringeTex" OutName="tunic_fringe" Format="rgb5a1" Width="16" Height="16" Offset="0x3BC" />
```

File: src/overlays/actors/ovl_En_Ko/z_en_ko.h [hunk 18]
```diff
@@ -6,11 +6,57 @@
...
+    /* 0x014C */ SkelAnime skelAnime;
+    /* 0x0190 */ EnKoActionFunc actionFunc;
+    /* 0x0194 */ s8 headObjectBankIdx;
+    /* 0x0195 */ s8 bodyObjectBankIdx;
+    /* 0x0196 */ s8 legsObjectBankIdx;
+    /* 0x0197 */ s8 osAnimeBankIndex;
+    /* 0x0198 */ ColliderCylinder collider;
+    /* 0x01E4 */ Path* path;
+    /* 0x01E8 */ struct_80034A14_arg1 unk_1E8;
+    /* 0X0210 */ u8 unk_210; //block trade quest sfx
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.h:22]:
```suggestion
    /* 0X0210 */ u8 unk_210; // block trade quest sfx
```

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 19]
```diff
@@ -1,21 +1,44 @@
...
+void func_80A99384(EnKo* this, GlobalContext* globalCtx);
+void func_80A99438(EnKo* this, GlobalContext* globalCtx);
+void func_80A99504(EnKo* this, GlobalContext* globalCtx);
+void func_80A99560(EnKo* this, GlobalContext* globalCtx);
+Gfx* EnKo_SetEnvColor(GraphicsContext* gfxCtx, u8 r, u8 g, u8 b, u8 a);
+s32 EnKo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx,
+                          Gfx** arg6);
+u16 func_80A97610(GlobalContext*, Actor*);
+s16 func_80A97738(GlobalContext*, Actor*);
+f32 func_80A97BC0(EnKo* this);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:33]: Most of these forward declarations look like they aren't necessary, so we would normally omit them

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 20]
```diff
@@ -47,93 +70,1257 @@ static ColliderCylinderInit D_80A9A100 = {
...
+
+typedef struct {
+    u8 headId;
+    u8 bodyId;
+    Color_RGBA8 tunicColor;
+    u8 legsId;
+    Color_RGBA8 bootsColor;
+} EnKoModelInfo;
+
+typedef enum { KO_BOY, KO_GIRL, KO_FADO } KokiriGender;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:163]: I suggest including offset comments in this enum to maintain a proper formatting. Also a lot of the other structs/enums declared here could use offset/size comments

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 21]
```diff
@@ -47,93 +70,1257 @@ static ColliderCylinderInit D_80A9A100 = {
...
+    /* ENKO_TYPE_CHILD_3    */ { 6, 30.0f, 180.0f },
+    /* ENKO_TYPE_CHILD_4    */ { 6, 30.0f, 180.0f },
+    /* ENKO_TYPE_CHILD_5    */ { 1, 30.0f, 240.0f },
+    /* ENKO_TYPE_CHILD_6    */ { 6, 30.0f, 180.0f },
+    /* ENKO_TYPE_CHILD_7    */ { 6, 30.0f, 180.0f },
+    /* ENKO_TYPE_CHILD_8    */ { 6, 30.0f, 180.0f },
+    /* ENKO_TYPE_CHILD_9    */ { 6, 30.0f, 180.0f },
+    /* ENKO_TYPE_CHILD_10   */ { 6, 30.0f, 180.0f },
+    /* ENKO_TYPE_CHILD_11   */ { 6, 30.0f, 180.0f },
+    /* ENKO_TYPE_CHILD_FADO */ { 6, 30.0f, 180.0f }
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:200]:
```suggestion
    /* ENKO_TYPE_CHILD_FADO */ { 6, 30.0f, 180.0f },
```

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 22]
```diff
@@ -47,93 +70,1257 @@ static ColliderCylinderInit D_80A9A100 = {
...
+    /* ENKO_TYPE_CHILD_3    */ { 0x00, 0x10, 0x10, 0x11, 0x12 },
+    /* ENKO_TYPE_CHILD_4    */ { 0x13, 0x13, 0x14, 0x0A, 0x09 },
+    /* ENKO_TYPE_CHILD_5    */ { 0x03, 0x03, 0x03, 0x03, 0x03 },
+    /* ENKO_TYPE_CHILD_6    */ { 0x04, 0x16, 0x16, 0x04, 0x17 },
+    /* ENKO_TYPE_CHILD_7    */ { 0x18, 0x10, 0x10, 0x19, 0x10 },
+    /* ENKO_TYPE_CHILD_8    */ { 0x1A, 0x0F, 0x0F, 0x1A, 0x0F },
+    /* ENKO_TYPE_CHILD_9    */ { 0x03, 0x03, 0x03, 0x1B, 0x1B },
+    /* ENKO_TYPE_CHILD_10   */ { 0x02, 0x02, 0x02, 0x02, 0x16 },
+    /* ENKO_TYPE_CHILD_11   */ { 0x0E, 0x0E, 0x0E, 0x0E, 0x0E },
+    /* ENKO_TYPE_CHILD_FADO */ { 0x05, 0x05, 0x05, 0x05, 0x05 }
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:152]:
```suggestion
    /* ENKO_TYPE_CHILD_FADO */ { 0x05, 0x05, 0x05, 0x05, 0x05 },
```

**Review by Roman971 — COMMENTED** [review]
Just a few things

**Comment by fig02** [comment]
changes made

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c [hunk 23]
```diff
@@ -25,12 +25,7 @@ void func_80A99384(EnKo* this, GlobalContext* globalCtx);
 void func_80A99438(EnKo* this, GlobalContext* globalCtx);
 void func_80A99504(EnKo* this, GlobalContext* globalCtx);
 void func_80A99560(EnKo* this, GlobalContext* globalCtx);
-Gfx* EnKo_SetEnvColor(GraphicsContext* gfxCtx, u8 r, u8 g, u8 b, u8 a);
-s32 EnKo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx,
-                          Gfx** arg6);
-u16 func_80A97610(GlobalContext*, Actor*);
-s16 func_80A97738(GlobalContext*, Actor*);
-f32 func_80A97BC0(EnKo* this);
+
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Ko/z_en_ko.c:28]:
```suggestion
```
empty line, fig cmon you had one job :feelsbadman:

File: tools/ZAPD/ZAPD/ZTexture.cpp [hunk 24]
```diff
@@ -835,13 +835,13 @@ TextureType ZTexture::GetTextureTypeFromString(std::string str)
 		texType = TextureType::RGBA32bpp;
 	else if (str == "rgba16")
 		texType = TextureType::RGBA16bpp;
-	else if (str == "rgb5a1")
+	else if (str == "rgba16")
```
Comment (Dragorn421) [inline tools/ZAPD/ZAPD/ZTexture.cpp:838]:
```suggestion
	else if (str == "rgb5a1")
```
monkaS

File: tools/ZAPD/ZAPD/ZTexture.cpp [hunk 25]
```diff
@@ -835,13 +835,13 @@ TextureType ZTexture::GetTextureTypeFromString(std::string str)
...
 	{
 		texType = TextureType::RGBA16bpp;
 #ifdef DEPRECATION_ON
 		fprintf(stderr,
 		        "ZTexture::GetTextureTypeFromString: Deprecation warning.\n"
-		        "\t The texture format 'rgb5a1' is currently deprecated, and will be removed in a future "
+		        "\t The texture format 'rgba16' is currently deprecated, and will be removed in a future "
```
Comment (Dragorn421) [inline tools/ZAPD/ZAPD/ZTexture.cpp:844]:
```suggestion
		        "\t The texture format 'rgb5a1' is currently deprecated, and will be removed in a future "
```
monkaS 2

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
what's this lazy ctrl+h (should have `git status`'d or `git add src`)

**Comment by fig02** [comment]
lmfao fuck, ty

File: src/overlays/actors/ovl_En_Ko/z_en_ko.c (hunk 23 above)
Comment (fig02) [reply src/overlays/actors/ovl_En_Ko/z_en_ko.c:28]: this was on purpose, the action funcs in a group makes sense to me

File: tools/ZAPD/ZAPD/ZTexture.cpp (hunk 24 above)
Comment (AngheloAlf) [reply tools/ZAPD/ZAPD/ZTexture.cpp:838]: Somebody forgot to exclude `tools/` from the global find and replace D:

Comment (Dragorn421) [reply tools/ZAPD/ZAPD/ZTexture.cpp:838]: fig is sabotaging zapd, you heard it here first

**Comment by Roman971** [comment]
Looks like the ZAPD readme was also accidentally modified

**Comment by fig02** [comment]
yup, already on it lmfao

**Comment by fig02** [comment]
ok I *think* its good now sorry lol
also i ran the formatter which touched some of the new audio stuff

**Review by Dragorn421 — APPROVED** [review]
Well this PR has seen a lot, first mzx's attempt to introduce rom assets data (scenes & rooms) into the repo, and now fig trying to sabotage zapd. Where are we headed?
EDIT: and jenkins failed to build damnit

**Comment by fig02** [comment]
ye today is not my day FeelsOkMan

**Comment by fig02** [comment]
ok now its actually ready lol

**Review by Roman971 — APPROVED** [review]

**Pull request merged**